File: depth-first-search-medium/881.救生艇.cpp
/*
 * @lc app=leetcode.cn id=881 lang=cpp
 *
 * [881] 救生艇
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution {
public:
    int numRescueBoats(vector<int> &people, int limit) {
        int ret = 0;
        sort(people.begin(), people.end());
        auto i = people.begin(), j = people.end() - 1;
        while (i <= j) {
            ret++;
            if (*i + *j <= limit)
                i++;
            j--;
        }
        return ret;
    }
};
// @lc code=end

File: depth-first-search-medium/881.救生艇.py
#
# @lc app=leetcode.cn id=881 lang=python3
#
# [881] 救生艇
#
from typing import List


# @lc code=start
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        l, r = 0, len(people)-1
        ret = 0
        while l <= r:
            ret += 1
            if people[l] + people[r] <= limit:
                l += 1
            r -= 1
        return ret
# @lc code=end
